Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/presencetracing/attendee/checkins/consent/CheckInsConsentViewModel.kt

```diff
@@ -33,17 +40,58 @@ class CheckInsConsentViewModel @AssistedInject constructor(
         }
     }.asLiveData(context = dispatcherProvider.Default)
 
+    val events = SingleLiveEvent<CheckInsConsentNavigation>()
+
     fun shareSelectedCheckIns() = launch {
+        Timber.d("Navigate to shareSelectedCheckIns")
+        autoSubmission.updateMode(AutoSubmission.Mode.MONITOR)
+
+        // Update CheckIns for submission
         val idsWithConsent = selectedSetFlow.value
         checkInRepository.updateSubmissionConsents(
             checkInIds = idsWithConsent,
             consent = true,
         )
-        // TODO persist selection and proceed to submitting check-ins
+
+        if (submissionRepository.hasViewedTestResult.first()) {
+            Timber.d("Navigate to SubmissionResultReadyFragment")
+            events.postValue(CheckInsConsentNavigation.ToSubmissionResultReadyFragment)
```

Comment:
use something like val event = when (....)
events.postValue(event)

Reply:
is not that already happening here ? or what do you mean?